aten/src/ATen/native/cuda/Normalization.cu

```diff
@@ -422,7 +422,7 @@ std::tuple<Tensor&, Tensor&, Tensor&> batch_norm_cuda_out(const Tensor& self, co
 }
 
 std::tuple<Tensor, Tensor, Tensor> batch_norm_cuda(const Tensor& self, const c10::optional<Tensor>& weight_opt, const c10::optional<Tensor>& bias_opt, const c10::optional<Tensor>& running_mean_opt, const c10::optional<Tensor>& running_var_opt, bool train, double momentum, double epsilon) {
-  auto output = at::empty_like(self, at::MemoryFormat::Contiguous);
+  auto output = at::empty_like(self, self.suggest_memory_format());
```

Comment:
can you also add a check in `batch_norm_elementwise` channels-last case that `output` is channels-last contiguous?

Reply:
cc @eqy

Reply:
followup in #62839